build_test.go

```diff
@@ -288,7 +288,7 @@ func testBuild(t *testing.T, context spec.G, it spec.S) {
 
 		context("when the layers directory cannot be written to", func() {
 			it.Before(func() {
-				Expect(os.Chmod(layersDir, 0000)).To(Succeed())
+				Expect(os.Chmod(layersDir, 4444)).To(Succeed())
```

Comment:
I believe the setuid flag (leading 4) has no effect. The setuid permission set on a **directory** is ignored. `4444` should be exactly same as `0444`. Can you change to `0444`?

Reply:
As far as I've been able to tell, the `0444` does not exercise the Reset() failure mode on neither macOS nor Linux. `4444` hits this failure mode on both.

Reply:
Hmm let me try to figure out why 0444 vs 4444 would have an impact before accepting this change

Reply:
Seems like the setting the setuid flag _can_ [have an effect](https://www.gnu.org/software/coreutils/manual/html_node/Directory-Setuid-and-Setgid.html) on directories, subsequently passing that setuid value to its files and subdirectories as well. This [file permission calculator](http://permissions-calculator.org/info/#setuid) also offered some insight for me.

Reply:
@thitch97 Thanks for the link. It makes sense and after looking around a bit, I agree that they have an effect on ubuntu and osx file systems but probably not very portable, just like the linked page says:
```
This behavior is a GNU extension. Portable scripts should not rely on requests to set or clear these bits on directories, as POSIX allows implementations to ignore these requests.
```

Reply:
Based on discussion in the analogous [PR to the dep buildpack](https://github.com/paketo-buildpacks/dep/pull/101) I decided to close this PR. Looks like it's a case of unclear test naming, not incomplete test coverage.